cmd/namedpipe.go

```diff
@@ -0,0 +1,190 @@
+package cmd
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"os"
+	"os/signal"
+	"syscall"
+
+	"github.com/go-logr/zapr"
+	"github.com/metal-toolbox/auditevent"
+	"github.com/metal-toolbox/auditevent/helpers"
+	"github.com/spf13/cobra"
+	"go.uber.org/zap"
+	"go.uber.org/zap/zapcore"
+	"golang.org/x/sync/errgroup"
+
+	"github.com/metal-toolbox/audito-maldito/ingesters/auditlog"
+	"github.com/metal-toolbox/audito-maldito/ingesters/namedpipe"
+	"github.com/metal-toolbox/audito-maldito/ingesters/rocky"
+	"github.com/metal-toolbox/audito-maldito/ingesters/syslog"
+	"github.com/metal-toolbox/audito-maldito/internal/common"
+	"github.com/metal-toolbox/audito-maldito/internal/health"
+	"github.com/metal-toolbox/audito-maldito/internal/metrics"
+	"github.com/metal-toolbox/audito-maldito/internal/util"
+	"github.com/metal-toolbox/audito-maldito/processors/auditd"
+	"github.com/metal-toolbox/audito-maldito/processors/sshd"
+)
+
+var (
+	appEventsOutput   string
+	auditdLogFilePath string
+	sshdLogFilePath   string
+	metricsCfg        metricsConfig
+)
+
+var namedpipeCmd = &cobra.Command{
+	Use:   "namedpipe",
+	Short: "Uses rsyslog for ingestion writing logs to namedpipe audito-maldito reads from.",
+	Long: `Uses rsyslog for ingestion writing logs to namedpipe audito-maldito reads from.
+	 Pipes must be created prior to opening the pipes.`,
+	Run: func(cmd *cobra.Command, args []string) {
```

Comment:
use `cmd.Context()` instead of `context.Background()`, that'll capture the context that cobra actually initiates.

Reply:
done. updated in `cmd/namedpipe.go` too